File: content/code-examples/tested/javascript/driver/time-series/create-query/create-query-collection.snippet.aggregate.js
const pipeline = [
  { $match: { 'sensor.sensorId': 5578 } },
  {
    $group: {
      _id: { $dateTrunc: { date: '$time', unit: 'day' } },
      avgTemp: { $avg: '$temp' },
    },
  },
  { $sort: { avgTemp: -1 } },
];

const cursor = weather.aggregate(pipeline);

File: content/code-examples/tested/java/driver-sync/timeseries/CreateQuery/CreateQueryCollection.snippet.aggregate.java
// Create an aggregation pipeline
List<Document> pipeline = Arrays.asList(
        new Document("$match", new Document("sensor.sensorId", 5578)),
        new Document("$group", new Document("_id", new Document("$dateTrunc", new Document("date", "$time").append("unit", "day")))
                .append("avgTemp", new Document("$avg", "$temp"))),
        new Document("$sort", new Document("avgTemp", -1))
);

// Run the aggregation
AggregateIterable<Document> aggregationResults = weather.aggregate(pipeline);
